internal/kvrpc/batch.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 TiKV Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kvrpc
+
+import (
+	"github.com/tikv/client-go/v2/internal/locate"
+	"github.com/tikv/client-go/v2/tikvrpc"
+)
+
+// Batch is part of the mutation set that will be sent to tikv in a request.
+type Batch struct {
+	RegionID locate.RegionVerID
+	Keys     [][]byte
+	Values   [][]byte
+}
+
+// BatchResult wraps a Batch request's server response or an error.
+type BatchResult struct {
+	*tikvrpc.Response
+	Error error
+}
+
+// AppendBatches divides the mutation to be requested into Batches, ensuring that the total key-value length
```

Comment:
This comment declares that the size of each `Batch` will not larger than `limit`, but the code in L40 tells us a new `Batch` is created only if the previous `Batch` is larger or equal to `limit`.

Reply:
The comments sematic make sense. We may modify the code itself.

Reply:
I think the code is correct because we need to handle the case when a single key-value pair exceeds the limit. I'll update the comment.

Reply:
OK.